src/DvidManager.jsx

```diff
@@ -7,52 +7,105 @@ import PropTypes from 'prop-types';
 import React from 'react';
 import TextField from '@material-ui/core/TextField';
 
+const KEY_GRAYSCALE_SOURCE = 'NG_WORKSPACES-FOCUSED-PROOFREADING-GRAYSCALE-SOURCE';
+const KEY_SEGMENTATION_SOURCE = 'NG_WORKSPACES-FOCUSED-PROOFREADING-SEGMENTATION-SOURCE';
+
 export class DvidManager {
   // TODO: Switch to an enum: 0 = not initialized, 1 = initializing, 2 = initialized.
   initialized = 0;
 
   onInitCompleted = undefined;
 
-  segmentationURL = 'https://flyem.dvid.io/d925633ed0974da78e2bb5cf38d01f4d/segmentation';
+  grayscaleURL = 'dvid://https://flyem.dvid.io/ab6e610d4fe140aba0e030645a1d7229/grayscalejpeg';
+
+  segmentationURL = 'dvid://https://flyem.dvid.io/d925633ed0974da78e2bb5cf38d01f4d/segmentation';
 
   init = (onInitCompleted) => {
     if (this.initialized === 0) {
+      if (this.localStorageAvailable()) {
+        // If localStorage is available, use it to remember the URLs across sessions.
+        const g = localStorage.getItem(KEY_GRAYSCALE_SOURCE);
+        if (g) {
+          this.grayscaleURL = g;
+        }
+        const s = localStorage.getItem(KEY_SEGMENTATION_SOURCE);
+        if (s) {
+          this.segmentationURL = s;
+        }
+      }
       this.onInitCompleted = onInitCompleted;
       this.initialized = 1;
     }
   }
 
+  grayscaleSourceURL = () => (
+    this.grayscaleURL
+  )
+
+  segmentationSourceURL = () => (
+    this.segmentationURL
+  )
+
+  sparseVolSize = (bodyId, onCompletion) => {
+    const url = `${this.segmentationAPIURL()}/sparsevol-size/${bodyId}`;
+    fetch(url).then((response) => response.json()).then((data) => onCompletion(data));
+  }
+
+  bodyId = (bodyPt, onCompletion) => {
+    const key = `${bodyPt[0]}_${bodyPt[1]}_${bodyPt[2]}`;
+    const url = `${this.segmentationAPIURL()}/label/${key}`;
+    fetch(url).then((response) => response.json()).then((data) => onCompletion(data.Label));
+  };
+
+  //
+
   onDialogClosed = () => {
     this.onInitCompleted();
   }
 
-  grayscaleSourceURL = () => {
-    // TODO: Get this value from the proper UI.
-    const result = 'dvid://https://flyem.dvid.io/ab6e610d4fe140aba0e030645a1d7229/grayscalejpeg';
-    return (result);
+  onGrayscaleSourceChange = (event) => {
+    this.grayscaleURL = event.target.value;
+    if (this.localStorageAvailable()) {
+      localStorage.setItem(KEY_GRAYSCALE_SOURCE, this.grayscaleURL);
+    }
   }
 
-  segmentationSourceURL = () => {
-    // TODO: Get this value from the proper UI.
-    const result = `dvid://${this.segmentationURL}`;
-    return (result);
+  onSegmentationSourceChange = (event) => {
+    this.segmentationURL = event.target.value;
+    if (this.localStorageAvailable()) {
+      localStorage.setItem(KEY_SEGMENTATION_SOURCE, this.segmentationURL);
+    }
   }
 
   segmentationAPIURL = () => {
```

Comment:
@hubbardp, have you looked at the URL() object for performing these types of functions on a url string? https://developer.mozilla.org/en-US/docs/Web/API/URL. It might be more robust.

Reply:
I see now that these are non-standard urls, so that is probably why you can use the standard url parsing